--- gtkterm-0.99.5/src/macros.c~	2006-03-31 22:04:21.000000000 +0200
+++ gtkterm-0.99.5/src/macros.c	2006-03-31 22:04:21.000000000 +0200
@@ -59,7 +59,7 @@
   guchar a;
   guint val_read;
   
-  string = macros[(gint)number].action;
+  string = macros[(long)number].action;
   length = strlen(string);
   
   for(i = 0; i < length; i++)
@@ -140,7 +140,7 @@
 	}
     }
 
-  str = g_strdup_printf(_("Macro \"%s\" sent !"), macros[(gint)number].shortcut);
+  str = g_strdup_printf(_("Macro \"%s\" sent !"), macros[(long)number].shortcut);
   Put_temp_message(str, 800);
   g_free(str);
 }
@@ -160,7 +160,7 @@
 
 void add_shortcuts(void)
 {
-  gint i = 0;
+  long i = 0;
   guint acc_key;
   GdkModifierType mod;
 
--- gtkterm-0.99.5/src/fichier.c~	2006-03-31 22:02:49.000000000 +0200
+++ gtkterm-0.99.5/src/fichier.c	2006-03-31 22:02:49.000000000 +0200
@@ -153,7 +153,7 @@
   else
     {
       g_free(str);
-      str = g_strdup_printf(_("Cannot read file %s : %s\n"), NomFichier, strerror_utf8(errno));
+      str = g_strdup_printf(_("Cannot read file %s : %s\n"), NomFichier, strerror(errno));
       show_message(str, MSG_ERR);
     }
   return FALSE;
@@ -202,7 +202,7 @@
 	{
 	  /* Problem while writing, stop file transfer */
 	  g_free(str);
-	  str = g_strdup_printf(_("Cannot write file %s\n"), strerror_utf8(errno));
+	  str = g_strdup_printf(_("Cannot write file %s\n"), strerror(errno));
 	  show_message(str, MSG_ERR);
 	  close_all();
 	  return;
@@ -298,7 +298,7 @@
   if(Fic == NULL)
     {
       g_free(str);
-      str = g_strdup_printf(_("Cannot open file %s : %s\n"), NomFichier, strerror_utf8(errno));
+      str = g_strdup_printf(_("Cannot open file %s : %s\n"), NomFichier, strerror(errno));
       show_message(str, MSG_ERR);
     }
   else
