--- gtkterm-0.99.5/src/widgets.c.orig	2007-11-21 22:22:25.000000000 +0100
+++ gtkterm-0.99.5/src/widgets.c	2007-11-21 22:23:23.000000000 +0100
@@ -581,12 +581,12 @@
   guchar val;
   guint val_read;
   guint sent = 0;
-  gchar written[3];
+  gchar written[4];
   gchar *all_written;
 
   text = (gchar *)gtk_entry_get_text(GTK_ENTRY(widget));
 
-  all_written = g_malloc(strlen(text) * 2);
+  all_written = g_malloc(strlen(text) * 2 + 1);
   all_written[0] = 0;
 
   current = text;
