--- src/widgets.c.str	2009-05-13 03:42:57.000000000 +0200
+++ src/widgets.c	2009-05-13 03:43:12.000000000 +0200
@@ -559,12 +559,12 @@
  if(type_msg==MSG_ERR)
    {
      Fenetre_msg = 
-       gtk_message_dialog_new(GTK_WINDOW(Fenetre), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, message);
+       gtk_message_dialog_new(GTK_WINDOW(Fenetre), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, "%s", message);
    }
  else if(type_msg==MSG_WRN)
    {
      Fenetre_msg = 
-       gtk_message_dialog_new(GTK_WINDOW(Fenetre), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, message);
+       gtk_message_dialog_new(GTK_WINDOW(Fenetre), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_WARNING, GTK_BUTTONS_OK, "%s", message);
    }
  else
    return;
